src/sig_stfl/sig_stfl.h

```diff
@@ -54,6 +54,74 @@ OQS_API int OQS_SIG_STFL_alg_count(void);
  */
 OQS_API int OQS_SIG_STFL_alg_is_enabled(const char *method_name);
 
+/**
+ * @brief OQS_SECRET_KEY object for stateful signature schemes
+ */
+typedef struct OQS_SECRET_KEY OQS_SECRET_KEY;
+
+typedef struct OQS_SECRET_KEY {
+    /** A local ordinal representing the LMS parameter of the signature scheme. */
+    uint32_t oid;
+
+    /** Printable string representing the name of the signature scheme. */
+    const char *method_name;
+
+    /* The (maximum) length, in bytes, of secret keys for this signature scheme. */
+    size_t length_secret_key;
+
+    /* The variant specific secret key data */
+    void *secret_key_data;
+
+    /* User defined data that may be used for the SAFETY functions */
+    void *data;
+
+    /* Function that returns the total number of signatures for the secret key */
+    unsigned long long (*sigs_total)(const OQS_SECRET_KEY *secret_key);
+
+    /* Function that returns the number of signatures left for the secret key */
+    unsigned long long (*sigs_left)(const OQS_SECRET_KEY *secret_key);
+
+    /**
+     * Secret Key retrieval Function
+     *
+     * @param[in] sk The secret key represented as OQS_SECRET_KEY object
+     * @return byte pointer or null
+     */
+    uint8_t * (*get_key)(OQS_SECRET_KEY *sk);
```

Comment:
Just to be sure about the convention here, the idea is that the function will malloc a new byte array of the appropriate size (presumably no bigger than length_secret_key up above) and then return a pointer to that?  Which the caller is responsible for freeing?  I think the API documentation could be clearer including indicating that the caller is responsible for freeing the result.  One thing that's unclear though is how the caller is told what the length of the allocated byte array is.

I think `get_key` is a bit unclear as a method name. Maybe either `key_to_byte_string` or `serialize_key`.

Also will need to add something that deserializes the key from a byte string back to an OQS_SECRET_KEY object.  Whether it's a static method of the OQS_SECRET_KEY object or a method of OQS_SIG_STFL.

Reply:
Changing to "serialize_key" and "deserializes". Will update and make clearer memory mgmt responsibilities.

   /**
     * Secret Key retrieval Function
     *
     * @param[in] sk The secret key represented as OQS_SECRET_KEY object
     * @param[out] key_len length of the returned byte string
     * @return byte pointer or null
     */
    uint8_t * (* serialize_key)(OQS_SECRET_KEY *sk, size_t *key_len);